sklearn/multioutput.py

```diff
@@ -359,3 +362,218 @@ def score(self, X, y):
                              format(n_outputs_, y.shape[1]))
         y_pred = self.predict(X)
         return np.mean(np.all(y == y_pred, axis=1))
+
+
+class ClassifierChain(BaseEstimator):
+    """A multi-label model that arranges binary classifiers into a chain.
+
+    Each model makes a prediction in the order specified by the chain using
+    all of the available features provided to the model plus the predictions
+    of models that are earlier in the chain.
+
+    Parameters
+    ----------
+    base_estimator : estimator
+        The base estimator from which the classifier chain is built.
+
+    random_state : int or RandomState, optional, default None
+        State or seed for random number generator.
+
+    order : list of integers, None, or 'random'
+        The order of the chain can be explicitly set by providing a list of
+        integers. For example, for a chain of length 5
+            order = [1, 3, 2, 4, 0]
+        means that the first model in the chain will make predictions for
+        column 1 in the Y matrix, the second model will make predictions
+        for column 3, etc.
+
+        If order is None the order will be determined by the
+        order of columns in the label matrix Y.
+            order = [0, 1, 2, ..., Y.shape[1] - 1]
+
+        If order is 'random' a random ordering will be used.
+
+    cv : int, cross-validation generator or an iterable, optional
+        Determines whether to use cross validated predictions or true
```

Comment:
Ok. Although I would point out that the original paper used exclusively binary true labels as features. Furthermore, in my experience using the cross validation predictions as features at training time consistently leads to worse performance. I don't have an explanation of this but it has been my experience. Nevertheless if you confirm that you still want this change I am willing to make it.